lib/puppet/provider/logical_volume/lvm.rb

```diff
@@ -36,6 +38,23 @@ def create
             args.push('--stripesize', @resource[:stripesize])
         end
 
+        if @resource[:mirror]
+            args.push('--mirrors', @resource[:mirror])
+            if @resource[:mirrorlog]
+                args.push('--mirrorlog', @resource[:mirrorlog])
+            end
+            if @resource[:region_size]
+                args.push('--regionsize', @resource[:region_size])
+            end
+            if @resource[:no_sync]
```

Comment:
"The *--nosync* option is an optimization - without it LVM will try
synchronize the mirror by copying empty sectors from one logical volume to
another."
It is a parameter to lvcreate. Why should it be used *without* mirrors?
See http://wiki.gentoo.org/wiki/LVM

  Martin Wangel



On Wednesday, February 4, 2015, Morgan Haskel <notifications@github.com>
wrote:

> In lib/puppet/provider/logical_volume/lvm.rb
> <https://github.com/puppetlabs/puppetlabs-lvm/pull/76#discussion_r24086955>
> :
>
> > @@ -36,6 +38,23 @@ def create
> >              args.push('--stripesize', @resource[:stripesize])
> >          end
> >
> > +        if @resource[:mirror]
> > +            args.push('--mirrors', @resource[:mirror])
> > +            if @resource[:mirrorlog]
> > +                args.push('--mirrorlog', @resource[:mirrorlog])
> > +            end
> > +            if @resource[:region_size]
> > +                args.push('--regionsize', @resource[:region_size])
> > +            end
> > +            if @resource[:no_sync]
>
> I don't see any indication that no_sync should only apply when mirror is
> set.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/puppetlabs/puppetlabs-lvm/pull/76/files#r24086955>.
>


-- 
-----------------------
 Martin Wangel
 Iteego